.circleci/config.yml

```diff
@@ -0,0 +1,123 @@
+      FCREPO_TEST_PORT: 8080/fcrepo
+      BUNDLE_JOBS: 4
+      BUNDLE_RETRY: 3
+      NOKOGIRI_USE_SYSTEM_LIBRARIES: true
+      ENGINE_CART_RAILS_OPTIONS: --skip-git --skip-bundle --skip-listen --skip-spring --skip-yarn --skip-keeps --skip-coffee --skip-puma --skip-test
+      COVERALLS_PARALLEL: true
+
+    steps:
+    - restore_cache:
+        keys:
+        - source-v1-{{ .Branch }}-{{ .Revision }}
+
+    - checkout
+
+    - save_cache:
+        key: source-v1-{{ .Branch }}-{{ .Revision }}
+        paths:
+        - ".git"
+
+    # BUNDLE_PATH is unset to allow for `bundle config path` to take precedence.
+    - run:
+        name: Extra environment setup
+        command: |
+          echo 'unset BUNDLE_PATH' >> $BASH_ENV
+
+    - restore_cache:
+        keys:
+        - v1-internal-test-app-{{ checksum "hyrax-iiif_av.gemspec" }}-{{ checksum "spec/test_app_templates/lib/generators/test_app_generator.rb" }}-{{ checksum "lib/generators/hyrax/iiif_av/install_generator.rb" }}
+
+    # Download and cache dependencies
+    - restore_cache:
+        keys:
+        - v1-dependencies-{{ .Branch }}-{{ checksum "hyrax-iiif_av.gemspec" }}-{{ checksum "spec/test_app_templates/lib/generators/test_app_generator.rb" }}-{{ checksum "lib/generators/hyrax/iiif_av/install_generator.rb" }}
+        - v1-dependencies-{{ .Branch }}
+        - v1-dependencies-
+
+    - run:
+        name: Install dependencies
+        command: |
+          gem update --system
+          gem update bundler
+          bundle config path $CIRCLE_WORKING_DIRECTORY/vendor/bundle
+          bundle install
+
+    - run:
+        name: Generate test app, ensure top-level Gemfile.lock is valid
+        command: |
+          [ -e ./.internal_test_app ] || bundle exec rake engine_cart:generate
+
+    - save_cache:
+        paths:
+        - ./.internal_test_app
+        key: v1-internal-test-app-{{ checksum "hyrax-iiif_av.gemspec" }}-{{ checksum "spec/test_app_templates/lib/generators/test_app_generator.rb" }}-{{ checksum "lib/generators/hyrax/iiif_av/install_generator.rb" }}
+
+    - run:
+        name: Install test app dependencies
+        command: |
+          cd .internal_test_app
+          bundle install
+
+    - save_cache:
+        paths:
+        - ./vendor/bundle
+        key: v1-dependencies-{{ .Branch }}-{{ checksum "hyrax-iiif_av.gemspec" }}-{{ checksum "spec/test_app_templates/lib/generators/test_app_generator.rb" }}-{{ checksum "lib/generators/hyrax/iiif_av/install_generator.rb" }}
+
+    - run:
+        name: Load config into SolrCloud
+        command: |
+          cd .internal_test_app/solr/config
+          zip -1 -r solr_hyrax_config.zip ./*
+          curl -H "Content-type:application/octet-stream" --data-binary @solr_hyrax_config.zip "http://localhost:8985/solr/admin/configs?action=UPLOAD&name=hyrax"
+          curl -H 'Content-type: application/json' http://localhost:8985/api/collections/ -d '{create: {name: hydra-test, config: hyrax, numShards: 1}}'
+
+    - run:
```

Comment:
Let's either setup code climate the rest of the way or take out this pre-build cc command.